### PR TITLE
Created unit tests for ObjectDependencyManager

### PR DESCRIPTION
**Pull request by Karnah**
I hope that this tests cover all conditions inside class. Also I am a bit extend checks inside `AddObjectDependency` method.

I've used `xUnit`, but I think it will be easy to change another system.
Also I've turned off parallelism, because `ObjectDependencyManager` is singleton and there will many problem if try run tests for parallel.

### CODE REVIEW

**Comment by Karnah** [comment]
I saw problem with CodeFactor. I use double space to separate blocks of arrange/act/assert inside method and separate group of methods (instead using regions). If this is a problem I'll fix it.

**Comment by Karnah** [comment]
@konne, I fixed all problems

File: tests/XAMLMarkupExtensions.UnitTests/ObjectDependencyManagerTests.cs [hunk 1]
```diff
@@ -123,24 +131,109 @@ public void AddObjectDependency_DependencyAlive_TargetAlive()
...
+            Assert.False(firstDependencyWeakReference.IsAlive);
+            
+            // First target should be still alive, because ObjectDependencyManager doesn't know that dependency is dead.
+            Assert.True(firstTargetWeakReference.IsAlive);
+            
+            // Register second target and its dependency. After that should cleaned first target. 
+            var isSecondRegistered = ObjectDependencyManager.AddObjectDependency(secondDependencyWeakReference, secondTarget);
+            
+            // Force collect first target.
+            GC.Collect();
```
Comment (konne) [inline tests/XAMLMarkupExtensions.UnitTests/ObjectDependencyManagerTests.cs:210]: @Karnah I'm not shure that an simple GC.Collect always enforce the full delete.

Comment (Karnah) [reply tests/XAMLMarkupExtensions.UnitTests/ObjectDependencyManagerTests.cs:210]: GC.Collect() collect all generations and use Default collection mode (which equals to GCCollectionMode.Forced).
There is no objects with finalizator, so it seems to me that it's really work.

**Pull request merged**